### PR TITLE
docs for creating order with billing agreement

### PR DESCRIPTION
**Pull request by kgangineni**
### Summary of changes

 - add docs for using Billing Agreement with native checkout

 ### Checklist

 - ~[ ] Added a changelog entry~

- @kgangineni

### CODE REVIEW

File: docs/PayPalNativeCheckout/README.md [hunk 1]
```diff
@@ -10,11 +10,13 @@ Follow these steps to add PayPal Native Checkout payments:
 
 ## Setup a PayPal Developer Account
 
-You will need to set up authorization to use the PayPal Payments SDK.
-Follow the steps in [Get Started](https://developer.paypal.com/api/rest/#link-getstarted) to create a client ID and secret, to generate access tokens.
+You will need to set up authorization to use the PayPal Payments SDK. Follow the steps
+in [Get Started](https://developer.paypal.com/api/rest/#link-getstarted) to create a client ID and
+secret, to generate access tokens.
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
You will need to set up authorization to use the PayPal Payments SDK. Follow the steps
in [Get Started](https://developer.paypal.com/api/rest/#link-getstarted) to create a client ID and
secret, needed to generate an access tokens.
```

File: docs/PayPalNativeCheckout/README.md [hunk 2]
```diff
@@ -10,11 +10,13 @@ Follow these steps to add PayPal Native Checkout payments:
...
 
-You will need a server integration to create an order to capture funds using the [PayPal Orders v2 API](https://developer.paypal.com/docs/api/orders/v2).
-For initial setup, the `curl` commands below can be used as a reference for making server-side RESTful API calls.
+You will need a server integration to create an order to capture funds using
+the [PayPal Orders v2 API](https://developer.paypal.com/docs/api/orders/v2). For initial setup,
+the `curl` commands below can be used as a reference for making server-side RESTful API calls.
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
You will need a server-side integration to create an order using
the [PayPal Orders v2 API](https://developer.paypal.com/docs/api/orders/v2). This order will be used to authorize or capture funds. For initial setup,
the `curl` commands below can be used as a reference for making server-side RESTful API calls.
```

File: docs/PayPalNativeCheckout/README.md [hunk 3]
```diff
@@ -32,10 +34,14 @@ dependencies {
 
 In order to integrate PayPalNative checkout, you will need:
 
-1. An app client ID and corresponding secret. This is to generate and access token that will allow you to create payment tokens, capture funds and authorize customers to place orders.
+1. An app client ID and corresponding secret. This is to generate and access token that will allow
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
1. A PayPal client ID and corresponding secret. This is to generate an access token that will allow
```

File: docs/PayPalNativeCheckout/README.md [hunk 4]
```diff
@@ -32,10 +34,14 @@ dependencies {
...
+   you to create payment tokens, capture funds and authorize customers to place orders.
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]: general question, is payment tokens the correct term? I know on Braintree we use nonce (which is confusing) but want to make sure we are referencing single use payments however we have deemed is correct for this SDK.

File: docs/PayPalNativeCheckout/README.md [hunk 5]
```diff
@@ -32,10 +34,14 @@ dependencies {
...
 2. Setting a return URL.
 
-A return URL is required for redirecting users back to the app after authenticating. Please reference our [developer documentation](https://developer.paypal.com/docs/business/native-checkout/android/) to create said url and also to learn about how to create a new PayPal application as well.
+A return URL is required for redirecting users back to the app after authenticating. Please
+reference
+our [developer documentation](https://developer.paypal.com/docs/business/native-checkout/android/)
+to create said url and also to learn about how to create a new PayPal application as well.
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]: Looks like the spacing got a little off above with reference on it's own line.

```suggestion
to create a return URL and also to learn about how to create a new PayPal application.
```

File: docs/PayPalNativeCheckout/README.md [hunk 6]
```diff
@@ -58,36 +65,38 @@ Set a listener on the client to receive payment flow callbacks:
...
 
-    override fun onPayPalCheckoutStart() {
-        // the paypal paysheet is about to appear
-    }
-    
-    override fun onPayPalSuccess(result: PayPalNativeCheckoutResult) {
-       // order was successfully approved and is ready to be captured/authorized (see step 6)
-    }
+   override fun onPayPalCheckoutStart() {
+      // the paypal paysheet is about to appear
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
      // the PayPal paysheet is about to appear
```

File: docs/PayPalNativeCheckout/README.md [hunk 7]
```diff
@@ -58,36 +65,38 @@ Set a listener on the client to receive payment flow callbacks:
...
-    }
+   override fun onPayPalCanceled() {
+      // the user canceled the flow
+   }
+
+   override fun onPayPalCheckoutShippingChange(
+      shippingChangeData: ShippingChangeData,
+      shippingChangeActions: ShippingChangeActions
+   ) {
+      // there has been a change in shipping address or shipping method. Patch the order accordingly
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
      // there has been a change in shipping address or shipping method, patch the order accordingly
```
Update to match the formatting of the other comments

File: docs/PayPalNativeCheckout/README.md [hunk 8]
```diff
@@ -148,13 +163,78 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+Follow steps here to [Initiate PayPal Native Checkout](#3-initiate-paypal-native-checkout)
+
 ## Test and go live
 
 ### 1. Test the PayPal integration
 
-Follow the [Create sandbox account](https://developer.paypal.com/api/rest/#link-createsandboxaccounts) instructions to create a PayPal test account.
-When prompted to login with PayPal during the payment flow on your mobile app, you can log in with the test account credentials created above to complete the Sandbox payment flow.
+Follow
+the [Create sandbox account](https://developer.paypal.com/api/rest/#link-createsandboxaccounts)
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]: This spacing looks off with just "follow" on it's own line

**Comment by kgangineni** [comment]
@jaxdesmarais fixed spacings & wording mentioned in comments

File: docs/PayPalNativeCheckout/README.md [hunk 9]
```diff
@@ -11,10 +11,10 @@ Follow these steps to add PayPal Native Checkout payments:
 ## Setup a PayPal Developer Account
 
 You will need to set up authorization to use the PayPal Payments SDK.
-Follow the steps in [Get Started](https://developer.paypal.com/api/rest/#link-getstarted) to create a client ID and secret, to generate access tokens.
+Follow the steps in [Get Started](https://developer.paypal.com/api/rest/#link-getstarted) to create a client ID and secret, needed to generate an access token.
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
Follow the steps in [Get Started](https://developer.paypal.com/api/rest/#link-getstarted) to create a client ID and secret. These values are needed to generate an access token.
```

File: docs/PayPalNativeCheckout/README.md [hunk 10]
```diff
@@ -11,10 +11,10 @@ Follow these steps to add PayPal Native Checkout payments:
...
 
-You will need a server integration to create an order to capture funds using the [PayPal Orders v2 API](https://developer.paypal.com/docs/api/orders/v2).
-For initial setup, the `curl` commands below can be used as a reference for making server-side RESTful API calls.
+You will need a server-side integration to create an order to capture funds using the [PayPal Orders v2 API](https://developer.paypal.com/docs/api/orders/v2).
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
The SDK requires a server-side integration to create an order to capture funds using the [PayPal Orders v2 API](https://developer.paypal.com/docs/api/orders/v2).
```

File: docs/PayPalNativeCheckout/README.md [hunk 11]
```diff
@@ -11,10 +11,10 @@ Follow these steps to add PayPal Native Checkout payments:
...
+This order will be used to authorize or capture funds. For initial setup, the `curl` commands below can be used as a reference for making server-side RESTful API calls.
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
The order created on your server will be used to authorize or capture funds. For initial setup, the `curl` commands below can be used as a reference for making server-side RESTful API calls.
```
We should specify where the funds are authorized or captured after `funds` above. In the SDK? Also on their server?

File: docs/PayPalNativeCheckout/README.md [hunk 12]
```diff
@@ -59,7 +59,7 @@ Set a listener on the client to receive payment flow callbacks:
 payPalClient.listener = object : PayPalNativeCheckoutListener {
 
     override fun onPayPalCheckoutStart() {
-        // the paypal paysheet is about to appear
+        // the PaPal paysheet is about to appear
```
Comment (jaxdesmarais) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
        // the PayPal paysheet is about to appear
```

File: docs/PayPalNativeCheckout/README.md [hunk 13]
```diff
@@ -148,6 +148,67 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
 --data-raw ''
 ```
 
+## Billing Agreement
```
Comment (jcnoriega) [inline docs/PayPalNativeCheckout/README.md:152]: Add a link to this section in the begining of the Readme

File: docs/PayPalNativeCheckout/README.md [hunk 14]
```diff
@@ -1,20 +1,21 @@
...
 The PayPal Native Checkout module in the PayPal SDK enables PayPal payments in your app
 
 Follow these steps to add PayPal Native Checkout payments:
 
 1. [Setup a PayPal Developer Account](#setup-a-paypal-developer-account)
-1. [Add PayPal Native Checkout Module](#add-paypal-native-checkout-module)
-1. [Test and go live](#test-and-go-live)
+2. [Add PayPal Native Checkout Module](#add-paypal-native-checkout-module)
+3. [Test and go live](#test-and-go-live)
+4. After initial setup, Follow instructions [here](#billing-agreement) for Billing Agreements
```
Comment (sshropshire) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
4. After initial setup, follow instructions [here](#billing-agreement) for Billing Agreements
```

File: docs/PayPalNativeCheckout/README.md [hunk 15]
```diff
@@ -148,6 +149,67 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
 --data-raw ''
 ```
 
+## Billing Agreement
```
Comment (sshropshire) [inline docs/PayPalNativeCheckout/README.md:152]:
```suggestion
## Billing Agreements
```

File: docs/PayPalNativeCheckout/README.md [hunk 16]
```diff
@@ -148,6 +149,67 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+    {
+      "return_url": "https://example.com/return",
+      "cancel_url": "https://example.com/cancel",
+      "notify_url": "https://example.com/notify",
+      "accepted_pymt_type": "INSTANT",
+      "skip_shipping_address": false,
+      "immutable_shipping_address": true
+    }
+  }
+}'
```
Comment (sshropshire) [inline docs/PayPalNativeCheckout/README.md:181]: Can we format the JSON indentation here to increase readability?

**Review by sshropshire — APPROVED** [review]

File: docs/PayPalNativeCheckout/README.md [hunk 17]
```diff
@@ -148,6 +149,67 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+}
+```
+
+### 2. Set Billing Agreement
+
+```kotlin
+paypalNativeClient.startCheckout(CreateOrder { createOrderActions ->
+   createOrderActions.setBillingAgreementId(billingAgreementId)
+})
+```
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md:207]: Use the same variable name so it is clear that the merchant should pass this token response value into the `setBillingAgreementId` method . I also made this comment on the[ iOS PR](https://github.com/paypal/iOS-SDK/pull/92).

```suggestion
```json
{
   "token_id": "<BILLING_AGREEMENT_TOKEN>"
}
```

### 2. Set Billing Agreement

```kotlin
paypalNativeClient.startCheckout(CreateOrder { createOrderActions ->
   createOrderActions.setBillingAgreementId("<BILLING_AGREEMENT_TOKEN>")
})
```
```

File: docs/PayPalNativeCheckout/README.md [hunk 18]
```diff
@@ -148,6 +149,67 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+{
+   "token_id": "<TOKEN>"
+}
+```
+
+### 2. Set Billing Agreement
+
+```kotlin
+paypalNativeClient.startCheckout(CreateOrder { createOrderActions ->
+   createOrderActions.setBillingAgreementId(billingAgreementId)
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md]: Use the same variable name so it is clear that the merchant should pass this token response value into the `setBillingAgreementId` method . I also made this comment on the[ iOS PR](https://github.com/paypal/iOS-SDK/pull/92).
```suggestion
   createOrderActions.setBillingAgreementId("<BILLING_AGREEMENT_TOKEN>")
```

File: docs/PayPalNativeCheckout/README.md [hunk 19]
```diff
@@ -148,6 +149,67 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+    }
+  }
+}'
+```
+
+**Response**
+
+```json
+{
+   "token_id": "<TOKEN>"
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
   "token_id": "<BILLING_AGREEMENT_TOKEN>"
```

File: docs/PayPalNativeCheckout/README.md [hunk 20]
```diff
@@ -148,6 +149,67 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+    "payment_method": "PAYPAL"
+  },
+  "plan":
+  {
+    "type": "MERCHANT_INITIATED_BILLING",
+    "merchant_preferences":
+    {
+      "return_url": "https://example.com/return",
+      "cancel_url": "https://example.com/cancel",
+      "notify_url": "https://example.com/notify",
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md]: Same comment I left in iOS PR - these should be removed from the create order call.

```suggestion
```

File: docs/PayPalNativeCheckout/README.md [hunk 21]
```diff
@@ -1,20 +1,21 @@
...
 The PayPal Native Checkout module in the PayPal SDK enables PayPal payments in your app
 
 Follow these steps to add PayPal Native Checkout payments:
 
 1. [Setup a PayPal Developer Account](#setup-a-paypal-developer-account)
-1. [Add PayPal Native Checkout Module](#add-paypal-native-checkout-module)
-1. [Test and go live](#test-and-go-live)
+2. [Add PayPal Native Checkout Module](#add-paypal-native-checkout-module)
+3. [Test and go live](#test-and-go-live)
+4. After initial setup, follow instructions [here](#billing-agreement) for Billing Agreements
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md]: Same comment as https://github.com/paypal/iOS-SDK/pull/92/files#r1025480436

File: docs/PayPalNativeCheckout/README.md (hunk 20 above)
Comment (kgangineni) [reply docs/PayPalNativeCheckout/README.md]: removed `return_url` 
this seems to be not updated https://developer.paypal.com/docs/api/payments.billing-agreements/v1/

**Review by scannillo — APPROVED** [review]

**Pull request merged**